src/Catalyst.Core.Modules.Dfs/WebApi/V0/Controllers/DagController.cs

```diff
@@ -0,0 +1,133 @@
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using MultiFormats;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Catalyst.Core.Modules.Dfs.WebApi.V0.Controllers
+{
+    /// <summary>
+    ///   A link to a CID.
+    /// </summary>
+    public class LinkedDataDto
+    {
+        /// <summary>
+        ///   The CID.
+        /// </summary>
+        [JsonProperty(PropertyName = "/")]
+        public string Link;
+    }
+
+    /// <summary>
+    ///   A CID as linked data.
+    /// </summary>
+    public class LinkedDataCidDto
+    {
+        /// <summary>
+        ///   A link to the CID.
+        /// </summary>
+        public LinkedDataDto Cid;
+    }
+
+    /// <summary>
+    ///   Manages the IPLD (linked data) Directed Acrylic Graph.
+    /// </summary>
+    public class DagController : IpfsController
+    {
+        /// <summary>
+        ///   Creates a new controller.
+        /// </summary>
+        public DagController(IDfsService dfs) : base(dfs) { }
+
+        /// <summary>
+        ///  Resolve a reference (NYI).
+        /// </summary>
+        [HttpGet, HttpPost, Route("dag/resolve")] // TODO
+        public Task Resolve(string arg)
+        {
+            throw new NotImplementedException("Resolving a dag reference is not implemented.");
+        }
+
+        /// <summary>
+        ///   Gets the content of some linked data.
+        /// </summary>
+        /// <param name="arg">
+        ///   A path, such as "cid", "/ipfs/cid/" or "cid/a".
+        /// </param>
+        [HttpGet, HttpPost, Route("dag/get")]
+        public async Task<JToken> Get(string arg) { return await IpfsCore.DagApi.GetAsync(arg, Cancel); }
+
+        /// <summary>
+        ///   Add some linked data.
+        /// </summary>
+        /// <param name="file">
+        ///   multipart/form-data.
+        /// </param>
+        /// <param name="cidBase">
+        ///   The base encoding algorithm.
+        /// </param>
+        /// <param name="format">
+        ///   The content type.
+        /// </param>
+        /// <param name="hash">
+        ///   The hashing algorithm.
+        /// </param>
+        /// <param name="pin">
+        ///   Pin the linked data.
+        /// </param>
+        [HttpPost("dag/put")]
+        public async Task<LinkedDataCidDto> Put(IFormFile file,
+            string format = "dag-cbor",
+            string hash = MultiHash.DefaultAlgorithmName,
+            bool pin = true,
+            [ModelBinder(Name = "cid-base")] string cidBase = MultiBase.DefaultAlgorithmName)
+        {
+            if (file == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)